File: src/types/todo.ts
export type Todo = {
	id: number;
	title: string;
	description: string;
	completed: boolean;
};

export type TodoRequest = {
	title: string;
	description: string;
	completed: boolean;
};

export type TodoResponse = {
	id: number;
	title: string;
	description: string;
	completed: boolean;
};

File: api/_database/models/todo.py
from sqlalchemy import Column, Integer, String, DateTime, func
from sqlalchemy.orm import relationship
from ..._database.db import Base
from sqlalchemy import Boolean, ForeignKey


class Todo(Base):
    __tablename__ = "todos"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, index=True)
    description = Column(String, index=True)
    completed = Column(Boolean, default=False)
    owner_id = Column(Integer, ForeignKey("users.id"))
    owner = relationship("User", back_populates="todos")
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now())
